spec/jobs/notify_job_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+RSpec.describe NotifyJob, type: :job do
+  subject(:perform!) { described_class.new.perform(notification_id: notification.id) }
+
+  let(:subscription) { create(:subscription) }
+  let(:notification) { create(:notification, subscription: subscription, delivered_at: nil, delivery_attempted_at: nil) }
+  let(:client) { class_double(Faraday, post: nil) }
+  let(:response) { instance_double(Faraday::Response, success?: true) }
+  let(:data) { ActiveModelSerializers::Adapter.create(NotificationSerializer.new(notification)).to_json }
+  let(:hmac) { Encryptor.hmac(notification.subscription.secret, data) }
+  let(:headers) { { 'PECS-SIGNATURE': hmac, 'PECS-NOTIFICATION-ID': notification.id } }
+
+  context 'when notification and subscription are active' do
+    before do
+      allow(Faraday).to receive(:new).and_return(client)
+      allow(client).to receive(:post).and_return(response)
+      perform!
+    end
+
+    context 'when notification is a success' do
+      it 'posts JSON to the callback_url' do
+        expect(client).to have_received(:post).with(notification.subscription.callback_url, data, headers)
+      end
+
+      it 'updates delivered_at' do
+        expect(notification.reload.delivered_at).not_to be_nil
+      end
+    end
+
+    context 'when notification is not a success' do
+      it 'updates delivery_attempts' do
```

Comment:
it is not clear from the context (probably it is implicit) why this is not a success, and what does actually mean "is not a success"

Reply:
it responds with `false` to the `success?` method

Reply:
hopefully it's clearer now

Reply:
ops sorry, I did not see the `success?: false` 
that's perfect!